server/book/orderpq.go

```diff
@@ -356,29 +399,23 @@ func (pq *OrderPQ) Reheap() {
 }
 
 // Insert will add an element, while respecting the queue's capacity.
-// if at capacity, fail
-// else (not at capacity)
-// 		- heap.Push, which is pq.Push (append at bottom) then heapup
+//   if (at capacity) or (have order already), fail
+//   else (not at capacity), push the order onto the heap
 func (pq *OrderPQ) Insert(ord *order.LimitOrder) bool {
 	pq.mtx.Lock()
 	defer pq.mtx.Unlock()
 
-	if pq.capacity == 0 {
-		return false
-	}
-
-	if ord == nil {
+	if pq.capacity == 0 || ord == nil {
```

Comment:
Why are we not `panic`king on `nil` `ord`?

Reply:
Both cases are certainly programmer error, but I'm having a hard time bringing myself to force killing the process.  I might start with a loud fmt.Println and we can convince ourselves it's ok.